lib/Controller/LoginController.php

```diff
@@ -509,6 +513,9 @@ public function code(string $state = '', string $code = '', string $scope = '',
 			$this->userSession->completeLogin($user, ['loginName' => $user->getUID(), 'password' => '']);
 			$this->userSession->createSessionToken($this->request, $user->getUID(), $user->getUID());
 			$this->userSession->createRememberMeToken($user);
+			// TODO server should/could be refactored so we don't need to manually create the user session and dispatch the login-related events
```

Comment:
Can you file an issue in server and reference that? Helps to keep track of other cases or considerations around that.